src/session/sessionReducer.js

```diff
@@ -162,26 +172,43 @@ export default (state: SessionState = initialState, action: Action): SessionStat
         ...state,
         needsInitialFetch: true,
         loading: false,
+
+        // The server told us that the old queue ID is invalid. Forget it,
+        // so we don't try to use it.
+        eventQueueId: null,
       };
 
     case LOGIN_SUCCESS:
       return {
         ...state,
         needsInitialFetch: true,
+
+        // We're about to request a new event queue; no use hanging on to
+        // any old one we might have.
+        eventQueueId: null,
       };
 
     case LOGOUT:
       return {
         ...state,
         needsInitialFetch: false,
         loading: false,
+
+        // Although it may be valid, forget the event queue ID -- we know
+        // the user won't care about it.
```

Comment:
I'd frame this a bit differently:
```suggestion
        // Stop polling this event queue.
```